src/ostorlab/serve_app/app.py

```diff
@@ -13,4 +16,13 @@ def create_app(path: str = "/graphql", **kwargs) -> flask.Flask:
             "graphql", schema=schema, **kwargs
         ),
     )
+
+    @app.before_request
+    def authenticate() -> Optional[tuple[flask.Response, int]]:
+        """Authenticate the request."""
+        api_key = flask.request.headers.get("X-API-KEY")
+        if api_key is None or models.APIKey.is_valid(api_key) is False:
+            return flask.jsonify({"error": "Unauthorized"}), 401
+        return None
```

Comment:
Would be nice to have a comment to state that returning none will trigger the standard flow to continue for someone who is not familiar with the Flask API.